modules/standard/IO.chpl

```diff
@@ -2985,33 +2985,40 @@ deprecated "lines with a local_style argument is deprecated"
 proc file.lines(param locking:bool = true, start:int(64) = 0,
                 end:int(64) = max(int(64)), hints:iohints=IOHINT_NONE,
                 in local_style:iostyle) throws {
-  return this.linesHelper(locking, start, end, new ioHintSet(hints),
+  return this.linesHelper(locking, start..end, new ioHintSet(hints),
                           local_style: iostyleInternal);
 }
 
+pragma "last resort"
+deprecated "file.lines with a start and/or end argument is deprecated.  Please use the new region argument instead"
+proc file.lines(param locking:bool = true, start:int(64) = 0,
+                end:int(64) = max(int(64)),
+                hints = ioHintSet.empty) throws {
+  return this.lines(locking, start..end, hints);
+}
+
 /* Iterate over all of the lines in a file.
 
    :returns: an object which yields strings read from the file
 
    :throws SystemError: Thrown if an object could not be returned.
  */
-proc file.lines(param locking:bool = true, start:int(64) = 0,
-                end:int(64) = max(int(64)),
+proc file.lines(param locking:bool = true, region: range(?) = 0..,
```

Comment:
If this isn't a suitable time to do so, we should open an issue to document these arguments. Or open a general issue about IO module documentation tasks and add this as an entry.

Reply:
Thought about doing that as well,  I've added it to https://github.com/chapel-lang/chapel/issues/18522 for now